### PR TITLE
Hierarchical nav

### PR DESCRIPTION
**Pull request by gvonkoss**
SO. This is the mass pivot point—it integrates some parts of `o-hierarchical-nav`, meaning that now we can add a dropdown menu to each primary nav item. This gets pulled into the drawer. 

Readme and migrations guide to follow when v3 is ready to go.

### CODE REVIEW

File: test/helpers/fixtures.js [hunk 1]
```diff
@@ -1,49 +1,83 @@
...
-						</a>
-					</li>
-				</ul>
-				<button class="o-header-services__scroll-button o-header-services__scroll-button--left" aria-hidden="true" disabled></button>
-				<button class="o-header-services__scroll-button o-header-services__scroll-button--right" aria-hidden="true" disabled></button>
-			</div>
+			<<ul class="o-header-services__primary-nav-list">
+				<li data-o-header-services-level="1">
+					<a href="#">Fruit</a>
+					<ul data-o-header-services-level="2">
```
Comment (notlee) [inline test/helpers/fixtures.js]: Accessibility needs a little more work. The submenus are present at all times, and there's no indication that clicking the top level nav should open them up. Sorry if I misunderstood your `aria-expanded` chat earlier.

Comment (notlee) [reply test/helpers/fixtures.js]: Gah, it doesn't look like there are great options though. 

I wonder if we can speak to Tom about making the link a link, and make the arrows a button which opens the subnav, similar to o-header.
screenshot 2019-01-23 at 17 16 47

File: src/scss/_drop-down.scss [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+				opacity 0.01s linear,
+				transform 0.2s ease-in;
+			z-index: -1;
+
+			&.o-header-services__list--right {
+				right: 0;
+			}
+		}
+
+		.core li[data-o-header-services-level]:hover,
```
Comment (notlee) [inline src/scss/_drop-down.scss]: 😂 kapture 2019-01-23 at 17 26 06

Maybe the arrows can be hidden for core?

File: src/scss/_drop-down.scss [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+				ul[data-o-header-services-level] {
+					opacity: 1;
+					position: relative;
+					z-index: 2;
+				}
+			}
+		}
+	}
+
+	.core .o-header-services {
```
Comment (notlee) [inline src/scss/_drop-down.scss]: Same here I guess

File: src/scss/_drop-down.scss (hunk 2 above)
Comment (gvonkoss) [reply src/scss/_drop-down.scss]: My mistake, the menu is supposed to drop down on hover for core. its fixed now.

File: src/scss/_drop-down.scss (hunk 3 above)
Comment (gvonkoss) [reply src/scss/_drop-down.scss]: see above

**Pull request merged**